Cant search Russian word

When i search any russian word, i get nothing.

ex:
/source/search?project=BalanceStorage&q=*%D0%B8%D1%85%D0%B0%D0%B8%D0%BB*&defs=&refs=&path=&hist=&type=
result:
Your search full:*ихаил* did not match any files. 

in XREF file with this word:
`
<a class="l" name="4" href="#4">4</a>Changeset 1499224 | TFS#1360809 | Checked in by &#1050;&#1091;&#1087;&#1095;&#1091;&#1082; &#1052;&#1080;&#1093;&#1072;&#1080;&#1083; &#1043;&#1077;&#1086;&#1088;&#1075;&#1080;&#1077;&#1074;&#1080;&#1095; | Balance recalculation procedure updated to support big intermediate values during calculation process`

Original file in UTF8
4 Changeset 1499224 | TFS#1360809 | Checked in by Купчук Михаил Георгиевич | Balance recalculation procedure updated to support big intermediate values during calculation process

versions:
OpenGrok-1.0
oracle-java8 8u121-1~webupd8~0          
tomcat8  8.5.14-1+deb9u1 
SMP Debian 4.9.30-2+deb9u2 (2017-06-26)

1st try - get error 
`
 [javac] /opt/opengrok/opengrok-git/OpenGrok-master/generatedsrc/org/opensolaris/opengrok/analysis/plain/PlainFullTokenizer.java:751: error: incompatible types: boolean cannot be converted to int
    [javac]                         return true;
`
change  true to 1

2. 
`
 [javac] /opt/opengrok/opengrok-git/OpenGrok-master/src/org/opensolaris/opengrok/analysis/FileAnalyzer.java:187: error: constructor PlainSymbolTokenizer in class PlainSymbolTokenizer cannot be applied to given types;
    [javac]     protected JFlexTokenizer SymbolTokenizer=new PlainSymbolTokenizer();
`

I think the best way for me to wait for a full PR

--- a/src/org/opensolaris/opengrok/analysis/plain/PlainFullTokenizer.lex
+++ b/src/org/opensolaris/opengrok/analysis/plain/PlainFullTokenizer.lex
@@ -41,7 +41,7 @@ super(in);
 %char
 
 //WhiteSpace     = [ \t\f\r]+|\n
-Identifier = [a-zA-Z_] [a-zA-Z0-9_]*
+Identifier = [a-zA-Z\p{Letter}_] [a-zA-Z\p{Letter}0-9\p{Number}_]*
 Number = [0-9]+|[0-9]+\.[0-9]+| "0[xX]" [0-9a-fA-F]+
 Printable = [\@\$\%\^\&\-+=\?\.\:]
 

--- a/src/org/opensolaris/opengrok/search/context/HistoryLineTokenizer.lex
+++ b/src/org/opensolaris/opengrok/search/context/HistoryLineTokenizer.lex
@@ -96,7 +96,7 @@ stopset.add( "1.0");
   }
 %}
 
-Identifier = [a-zA-Z_] [a-zA-Z0-9_]*
+Identifier = [a-zA-Z\p{Letter}_] [a-zA-Z\p{Letter}0-9\p{Number}_]*
 Number = [0-9]+|[0-9]+\.[0-9]+| "0[xX]" [0-9a-fA-F]+
 Printable = [\@\$\%\^\&\-+=\?\.\:]
 

--- a/src/org/opensolaris/opengrok/search/context/PlainLineTokenizer.lex
+++ b/src/org/opensolaris/opengrok/search/context/PlainLineTokenizer.lex
@@ -394,7 +394,7 @@ import org.opensolaris.opengrok.analysis.Scopes.Scope;
 %}
 
 //WhiteSpace     = [ \t\f\r]+|\n
-Identifier = [a-zA-Z_] [a-zA-Z0-9_]*
+Identifier = [a-zA-Z\p{Letter}_] [a-zA-Z\p{Letter}0-9\p{Number}_]*
 Number = [0-9]+|[0-9]+\.[0-9]+| "0[xX]" [0-9a-fA-F]+
 Printable = [\@\$\%\^\&\-+=\?\.\:]
 

